### PR TITLE
Fix the crates publishing process

### PR DESCRIPTION
**Pull request by VanBarbascu**
Tested with:
```
cargo ws exec -- cargo check --all-features
cargo ws exec -- cargo check --no-default-features
cargo ws exec -- cargo check      # This is the one that we needs to be pass. Others are nice to have for now.
cargo ws publish --dry-run
cargo build -pneard
```

### CODE REVIEW

**Review by nagisa — APPROVED** [review]

File: core/primitives/Cargo.toml
Comment (frol) [inline core/primitives/Cargo.toml:92]: Does `clock` really need to be enabled by default?

It pulls in `tokio` as a dependency:

https://github.com/near/nearcore/blob/1eeee00448cff71c1f6b994f63fae009f9dfc788/core/time/Cargo.toml#L14-L24

Comment (akorchyn) [reply core/primitives/Cargo.toml:92]: not sure if even rand is mandatory ...

https://github.com/near/nearcore/commit/04d2b4f7e3c2882f8873f52bd6ee5355eb5b298b this PR introduces default-less

Comment (VanBarbascu) [reply core/primitives/Cargo.toml:92]: I am open to suggestions. So far, if I remove `clock`, `cargo ws exec -- cargo check` will fail on
```
Checking near-primitives v0.0.0                                                                                                                             
error[E0433]: failed to resolve: could not find `serde_utc_as_iso` in `near_time`                                                                                                                                   
   --> core/primitives/src/views.rs:337:20                                                                                                                                                                          
    |                                                                                                                                                                                                               
337 |     #[serde(with = "near_time::serde_utc_as_iso")]                                                                                                                                                            
    |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ could not find `serde_utc_as_iso` in `near_time`                                                                                                             
                                                                                                               
```
If I add "near-time/serde" then it fails on 
```
--> core/dyn-configs/src/lib.rs:6:5
  |
6 | use near_time::Clock;
  |     ^^^^^^^^^^^^^^^^ no `Clock` in the root
```
so we need both deps that are included under `clock`.

Comment (akorchyn) [reply core/primitives/Cargo.toml:92]: probably, somebody added without feature-gates :(

we need to restrict it futher

Comment (VanBarbascu) [reply core/primitives/Cargo.toml]: I added another commit.

Comment (akorchyn) [reply core/primitives/Cargo.toml]:
```diff
diff --git a/core/dyn-configs/Cargo.toml b/core/dyn-configs/Cargo.toml
index 15f68df4d..ccbffb4ec 100644
--- a/core/dyn-configs/Cargo.toml
+++ b/core/dyn-configs/Cargo.toml
@@ -20,7 +20,7 @@ thiserror.workspace = true
 tokio.workspace = true
 tracing.workspace = true
 
-near-time.workspace = true
+near-time = { workspace = true, features = ["clock"] }
 near-chain-configs.workspace = true
 near-crypto.workspace = true
 near-o11y.workspace = true
diff --git a/core/primitives/Cargo.toml b/core/primitives/Cargo.toml
index d592da655..a7cfdbbb2 100644
--- a/core/primitives/Cargo.toml
+++ b/core/primitives/Cargo.toml
@@ -58,43 +58,40 @@ rand = ["dep:rand", "rand_chacha", "near-crypto/rand", "itertools"]
 clock = ["near-time/clock", "near-time/serde"]
 dump_errors_schema = ["near-rpc-error-macro/dump_errors_schema"]
 protocol_feature_fix_staking_threshold = [
-  "near-primitives-core/protocol_feature_fix_staking_threshold",
+    "near-primitives-core/protocol_feature_fix_staking_threshold",
 ]
 protocol_feature_fix_contract_loading_cost = [
-  "near-primitives-core/protocol_feature_fix_contract_loading_cost",
+    "near-primitives-core/protocol_feature_fix_contract_loading_cost",
 ]
 protocol_feature_reject_blocks_with_outdated_protocol_version = [
-  "near-primitives-core/protocol_feature_reject_blocks_with_outdated_protocol_version",
+    "near-primitives-core/protocol_feature_reject_blocks_with_outdated_protocol_version",
 ]
 protocol_feature_nonrefundable_transfer_nep491 = [
-  "near-primitives-core/protocol_feature_nonrefundable_transfer_nep491",
+    "near-primitives-core/protocol_feature_nonrefundable_transfer_nep491",
 ]
 
 nightly = [
-  "near-fmt/nightly",
-  "near-parameters/nightly",
-  "near-primitives-core/nightly",
-  "near-primitives/nightly",
-  "nightly_protocol",
-  "protocol_feature_fix_contract_loading_cost",
-  "protocol_feature_fix_staking_threshold",
-  "protocol_feature_nonrefundable_transfer_nep491",
-  "protocol_feature_reject_blocks_with_outdated_protocol_version",
+    "near-fmt/nightly",
+    "near-parameters/nightly",
+    "near-primitives-core/nightly",
+    "near-primitives/nightly",
+    "nightly_protocol",
+    "protocol_feature_fix_contract_loading_cost",
+    "protocol_feature_fix_staking_threshold",
+    "protocol_feature_nonrefundable_transfer_nep491",
+    "protocol_feature_reject_blocks_with_outdated_protocol_version",
 ]
 
 nightly_protocol = [
-  "near-fmt/nightly_protocol",
-  "near-parameters/nightly_protocol",
-  "near-primitives-core/nightly_protocol",
-  "near-primitives/nightly_protocol",
+    "near-fmt/nightly_protocol",
+    "near-parameters/nightly_protocol",
+    "near-primitives-core/nightly_protocol",
+    "near-primitives/nightly_protocol",
 ]
 
-default = ["rand", "clock"]
 calimero_zero_storage = []
 
-protocol_schema = [
-  "near-schema-checker-lib/protocol_schema",
-]
+protocol_schema = ["near-schema-checker-lib/protocol_schema"]
 
 [dev-dependencies]
 chrono = { workspace = true, features = ["clock"] }
diff --git a/core/time/Cargo.toml b/core/time/Cargo.toml
index 408a22b98..f874b2678 100644
--- a/core/time/Cargo.toml
+++ b/core/time/Cargo.toml
@@ -20,5 +20,6 @@ serde = { workspace = true, optional = true }
 serde_json.workspace = true
 
 [features]
+default = ["serde"]
 clock = ["tokio"]
 serde = ["dep:serde", "time/serde"]
```

I think, this diff should make it working

Comment (akorchyn) [reply core/primitives/Cargo.toml]: You can apply it. Please, note thhat I have made on top of previous your commit

**Comment by VanBarbascu** [comment]
In this PR I was aiming to bring parity with the changes from the 2.1.1 release. Can we do this exploration in another PR? @akorchyn

**Comment by akorchyn** [comment]
CI won't pass as it checks for amount of crates ...

**Review by tayfunelmas — APPROVED** [review]

File: core/primitives/Cargo.toml
Comment (VanBarbascu) [reply core/primitives/Cargo.toml]: It worked! Thanks for the input!

**Comment by akorchyn** [comment]
Thanks for work on it !

By the way, you can participate in friendly competition 
@race-of-sloths invite

**Comment by race-of-sloths** [comment]
@akorchyn Thank you for calling!
    
@VanBarbascu Thank you for the contribution! Join [Race of Sloths](https://race-of-sloths.com) by simply mentioning me in your comment/PRs description and start collecting Sloth Points through contributions to open source projects.

<details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/VanBarbascu)

For maintainers:
- Score pull requests that participate in the Race of Sloths
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Invite contributor** `@race-of-sloths invite` to invite the contributor to participate in a race or include it, if it's already a runner.
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

**Pull request merged**
